### PR TITLE
Improve hiding secrets in logs

### PR DESCRIPTION
**Pull request by vitlibar**
### Changelog category:
- Improvement

### Changelog entry:
Improve hiding secrets in logs

### CODE REVIEW

**Comment by robot-ch-test-poll** [comment]
*This is an automated comment for commit cafe4429e3ee751ad9b43a723ad84c9a12d29bc9 with description of existing statuses. It's updated for the latest CI running*

[⏳ Click here](https://s3.amazonaws.com/clickhouse-test-reports/55089/cafe4429e3ee751ad9b43a723ad84c9a12d29bc9/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Mergeable Check</td><td>Checks if all other necessary checks are successful</td><td>✅ success</td></tr>
<tbody>
</table>
</details>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td>⏳ pending</td></tr>
<tbody>
</table>

**Comment by vitlibar** [comment]
@thevar1able Hi Konstantin, it would be nice if you review this PR :)

File: tests/queries/0_stateless/02801_backup_native_copy.reference [hunk 1]
```diff
@@ -1,4 +1,4 @@
-BACKUP TABLE data TO S3(s3_conn, \'backups/default/data_native_copy\') SETTINGS allow_s3_native_copy = 1	1
-BACKUP TABLE data TO S3(s3_conn, \'backups/default/data_no_native_copy\') SETTINGS allow_s3_native_copy = 0	0
-RESTORE TABLE data AS data_native_copy FROM S3(s3_conn, \'backups/default/data_native_copy\') SETTINGS allow_s3_native_copy = 1	1
-RESTORE TABLE data AS data_no_native_copy FROM S3(s3_conn, \'backups/default/data_no_native_copy\') SETTINGS allow_s3_native_copy = 0	0
+BACKUP TABLE data TO S3(s3_conn, \'backups/default/data_native_copy\') SETTINGS allow_s3_native_copy=true	1
```
Comment (thevar1able) [inline tests/queries/0_stateless/02801_backup_native_copy.reference:1]: Is this related to improving hiding secrets?

File: src/Parsers/ASTFunction.cpp [hunk 2]
```diff
@@ -150,41 +155,26 @@ namespace
...
-
-                /// The argument after 'url' is not a format so we do our replacement:
-                /// s3('url', 'aws_access_key_id', 'aws_secret_access_key', ...) -> s3('url', 'aws_access_key_id', '[HIDDEN]', ...)
-                markSecretArgument(url_arg_idx + 2);
+                    if (second_arg == "auto" || KnownFormatNames::instance().exists(second_arg))
+                        return; /// The argument after 'url' is a format: s3('url', 'format', ...)
+                }
             }
+
+            /// We're going to replace 'aws_secret_access_key' with '[HIDDEN'] for the following signatures:
```
Comment (thevar1able) [inline src/Parsers/ASTFunction.cpp:174]:
```suggestion
            /// We're going to replace 'aws_secret_access_key' with '[HIDDEN]' for the following signatures:
```

**Review by thevar1able — APPROVED** [review]
LGTM

File: tests/queries/0_stateless/02801_backup_native_copy.reference (hunk 1 above)
Comment (vitlibar) [reply tests/queries/0_stateless/02801_backup_native_copy.reference:1]: I changed the hiding algorithm a bit so now it returns the same query as it was if there was nothing to hide. I mean it used to replace `true` with `1`, and now it doesn't do that anymore.
